src/sagas/repository.ts

```diff
@@ -228,34 +228,49 @@ export function* createSnapshot(): any {
   const {
     name,
     description,
+    mode,
     assetName,
     filterStatement,
     joinStatement,
     policies,
   } = snapshots.snapshotRequest;
 
   const datasetName = dataset.name;
-  const mode = 'byQuery';
-  const selectedAsset = _.find(dataset.schema.assets, (asset) => asset.name === assetName);
+  let snapshotRequest;
```

Comment:
nit: Rather than a let with some assignments here, I'd love to try to use a switch expression on the `SnapshotRequestContentsModelModeEnum` - I think that will let us be a bit more declarative about things and maintain immutability